### PR TITLE
feat: [FFM-2675]: Git Sync & OPA Integration in Targets/Target Group Management

### PR DESCRIPTION
**Pull request by laura-mannus**
##### Summary:

When GitSync is enabled on a project, the user accesses the Target or Target Group details page the GitSync toolbar should be displayed. Therefore when a mutation is made against a flag on either the Target or Target Group details page then GitSync information should be collected and submitted with the mutation to Git.

##### Jira Links:

https://harness.atlassian.net/browse/FFM-2674

##### Screenshots:

Gitsync with Targets:

https://user-images.githubusercontent.com/103498167/176207782-f6dd4937-0ae4-40ed-8534-1920a42c8b34.mov

Gitsybc with Target Groups:

https://user-images.githubusercontent.com/103498167/176209629-3d9aa8aa-84e3-4c2b-ab46-a5f73c617682.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by laura-mannus** [comment]
retrigger typecheck

**Comment by laura-mannus** [comment]
retrigger standards

**Comment by laura-mannus** [comment]
retrigger typecheck

File: src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsPanel.tsx [hunk 1]
```diff
@@ -77,6 +82,18 @@ const FlagSettingsPanel: FC<FlagSettingsPanelProps> = ({ targetGroup }) => {
     }
   })
 
+  const handleSave = async (data: any): Promise<void> => {
```
Comment (chrisblakely01) [inline src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsPanel.tsx:85]: can we type this?

```suggestion
  const handleSave = async (data: PatchOperation): Promise<void> => {
```

File: src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsPanel.tsx [hunk 2]
```diff
@@ -87,38 +104,35 @@ const FlagSettingsPanel: FC<FlagSettingsPanelProps> = ({ targetGroup }) => {
...
-      targetGroup.identifier,
-      initialValues,
-      flags,
-      patchTargetGroup,
-      showSuccess,
-      refetch,
-      showError,
-      getRBACErrorMessage
-    ]
+    [targetGroup.identifier, initialValues, flags, saveWithGit, showSuccess, getString]
```
Comment (chrisblakely01) [inline src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsPanel.tsx:117]: do we need `handleSave` in here? it appears in some other places

File: src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/__tests__/FlagSettingsPanel.GitSync.test.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        repoIdentifier: repoDetails.repoIdentifier || 'harnesstest',
+        rootFolder: repoDetails.rootFolder || '/.harness/',
+        yamlError: repoDetails.yamlError || ''
+      },
+      repoSet: repoSet
+    }
+  } as any)
+}
+
+describe('Targeting Groups GitSync', () => {
```
Comment (chrisblakely01) [inline src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/__tests__/FlagSettingsPanel.GitSync.test.tsx:56]:
```suggestion
describe('FlagSettingsPanel GitSync', () => {
```

**Review by chrisblakely01 — COMMENTED** [review]
This is sweeet - Nice work, just a few things/questions

File: src/modules/75-cf/components/GitSyncActions/BranchSettingsButton.tsx [hunk 4]
```diff
@@ -38,6 +38,7 @@ const BranchSettingsButton = ({ isSettingsOpen, setIsSettingsOpen, isLoading }:
         onInteraction: nextOpenState => setIsSettingsOpen(nextOpenState)
       }}
       tooltip={<SettingsMenu />}
+      onClick={e => e.preventDefault()}
```
[the remark this answers is not in the document]
Comment (crystal-fong) [reply src/modules/75-cf/components/GitSyncActions/BranchSettingsButton.tsx:41]: yep that's right

**Comment by crystal-fong** [comment]
retrigger typecheck

File: src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/TargetManagementFlagConfigurationPanel.tsx [hunk 5]
```diff
@@ -159,14 +163,17 @@ const TargetManagementFlagConfigurationPanel: FC<TargetManagementFlagConfigurati
...
+            <div className={css.flaggitsync}>
+              <AddFlagButton
+                item={item}
+                onAdd={onAdd}
+                existingFlagIds={existingFlagIds}
+                includePercentageRollout={includePercentageRollout}
+                planEnforcementProps={planEnforcementProps}
+                title={addFlagsDialogTitle}
+              />
+              {isGitSyncActionsEnabled ? <TargetManagementToolbar /> : <div />}
```
[the remark this answers is not in the document]
Comment (crystal-fong) [reply src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/TargetManagementFlagConfigurationPanel.tsx:175]: done

File: src/modules/75-cf/hooks/useGitSync.tsx [hunk 6]
```diff
@@ -199,8 +200,8 @@ export const useGitSync = (): UseGitSync => {
   const [showGitSyncModal, hideGitSyncModal] = useModalHook(
     () => (
       <SaveFlagToGitModal
-        flagName={entityDataRef.current.featureFlagName}
-        flagIdentifier={entityDataRef.current.featureFlagIdentifier}
+        flagName={entityDataRef.current.featureFlagName ?? ''}
+        flagIdentifier={entityDataRef.current.featureFlagIdentifier ?? ''}
```
[the remark this answers is not in the document]
Comment (crystal-fong) [reply src/modules/75-cf/hooks/useGitSync.tsx:204]: oop yep, done

**Pull request closed without merging**